Matrix.py

```diff
@@ -0,0 +1,123 @@
+class Matrix:
```

Comment:
You should be uploading a file of unit tests as well. Call it test_matrix.py (and think more about what cases you need to test. Looking at your code I can see that you missed an important test case)